### PR TITLE
add request id in logs

### PR DESCRIPTION
**Pull request by XciD**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by Narsil** [comment]
LGTM actually

File: router/src/server.rs [hunk 1]
```diff
@@ -967,7 +970,35 @@ pub async fn run(
...
+                    },
+                )
+                .on_failure(
+                    |error: ::tower_http::classify::ServerErrorsFailureClass,
+                     latency: ::std::time::Duration,
+                     _span: &::tracing::Span| {
+                        ::tracing::warn!(took = latency.as_secs_f32(), "Failure: {error:?}");
+                    },
+                ),
+        )
```
Comment (OlivierDehaene) [inline router/src/server.rs:999]: Do we need this? We already have tracing in the server.
Can't the request id just be set by a parent span and we keep the current tracing logic?

File: router/src/server.rs [hunk 2]
```diff
@@ -1103,3 +1134,29 @@ impl From<InferError> for Event {
...
 }
+
+#[derive(Clone)]
+pub struct RequestSpan {}
+
+impl RequestSpan {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
```
Comment (OlivierDehaene) [inline router/src/server.rs:1145]: Is this needed?

```suggestion
```

**Review by OlivierDehaene — COMMENTED** [review]
We already have our own more precise tracing logic.
Is this compatible with axum_tracing_opentelemetry? 
Can't we add the request_id in a parent span without modifying the logging?

**Pull request closed without merging**